### PR TITLE
Update pushgateway type def

### PR DESCRIPTION
**Pull request by adilhafeez**
Update pushgateway typedef to reflect javascript implementation

```
const { resp, body } = await pushGateway.push({
  jobName: serviceName,
});
console.log(body);
console.log(`${resp.statusCode}`);

```

output
```

200
```

### CODE REVIEW

**Comment by adilhafeez** [comment]
ptal @PierrickP / @siimon

File: index.d.ts [hunk 1]
```diff
@@ -562,7 +562,7 @@ export class Pushgateway {
 	pushAdd(
 		params: Pushgateway.Parameters,
 		callback: (error?: Error, httpResponse?: any, body?: any) => void,
```
Comment (SimenB) [inline index.d.ts:564]: should remove all the callbacks as well

File: index.d.ts [hunk 2]
```diff
@@ -562,7 +562,7 @@ export class Pushgateway {
...
-	): void;
+	): Promise<{ resp: any, body: any }>;
```
Comment (SimenB) [inline index.d.ts:565]:
```suggestion
	): Promise<{ httpResponse?: any, body?: any }>;
```

just to match what we have.

We should also probably type them as `unknown` and not `any`

**Review by SimenB — CHANGES_REQUESTED** [review]

**Comment by PierrickP** [comment]
:\ indeed i did not update the typedef (sorry) as i'm not using it.
How can we test / unit test it for futurs update ?

**Comment by SimenB** [comment]
We could add `tsd` or something

File: index.d.ts (hunk 1 above)
Comment (adilhafeez) [reply index.d.ts:564]: removed

File: index.d.ts (hunk 2 above)
Comment (adilhafeez) [reply index.d.ts]: it seems like I cannot set resp to httpResponse - we'd have to change here too https://github.com/siimon/prom-client/blob/f177b1fd3d4db5fc48fcb1ec02d94069fffcf144/lib/pushgateway.js#L71

**Review by SimenB — APPROVED** [review]

**Comment by adilhafeez** [comment]
@siimon I added tsd-check but seems like github won't allow me to create another workflow

Here is the error I got,
```
 ! [remote rejected] fix_pushgateaway_typedef -> fix_pushgateaway_typedef (refusing to allow a Personal Access Token to create or update workflow `.github/workflows/nodejs.yml` without `workflow` scope)
```

Here is the patch,
```
diff --git a/.github/workflows/nodejs.yml b/.github/workflows/nodejs.yml
index 6d35657..6051440 100644
--- a/.github/workflows/nodejs.yml
+++ b/.github/workflows/nodejs.yml
@@ -40,3 +40,5 @@ jobs:
         run: npm i
       - name: Test
         run: npm run test
+      - name: TSD Check
+        run: npx tsd-check
diff --git a/index.test-d.ts b/index.test-d.ts
new file mode 100644
index 0000000..e28823b
--- /dev/null
+++ b/index.test-d.ts
@@ -0,0 +1,17 @@
+import { Pushgateway } from './index';
+
+const pushGateway = new Pushgateway('');
+pushGateway.push({ jobName: '' }).then(({ resp, body }) => {
+       console.log(resp);
+       console.log(body);
+});
+
+pushGateway.pushAdd({ jobName: '' }).then(({ resp, body }) => {
+       console.log(resp);
+       console.log(body);
+});
+
+pushGateway.delete({ jobName: '' }).then(({ resp, body }) => {
+       console.log(resp);
+       console.log(body);
+});
diff --git a/package.json b/package.json
index 1bd3f23..0da851b 100644
--- a/package.json
+++ b/package.json
@@ -45,7 +45,8 @@
                "typescript": "^4.0.2"
        },
        "dependencies": {
-               "tdigest": "^0.1.1"
+               "tdigest": "^0.1.1",
+               "tsd-check": "^0.6.0"
        },
        "types": "./index.d.ts",
        "jest": {
```

**Comment by adilhafeez** [comment]
@SimenB I missed changelog file. I've added it just now can you pls approve it again? (workflows are blocked)

**Comment by adilhafeez** [comment]
I don't seem to have permission to merge. One of the maintainers need to merge this PR? @SimenB could you help?

**Review by zbjornson — APPROVED** [review]

**Comment by Ghazgkull** [comment]
@zbjornson @SimenB The changes merged in this PR are great, but they still haven't been released. It seems that the package version didn't get bumped. The result is that the promise-based API has been broken/unavailable for Typescript users for some time now. Is it possible to cut a release with this fix please?

**Comment by SimenB** [comment]
https://github.com/siimon/prom-client/releases/tag/v14.0.1

**Pull request merged**